Create announcements.md (#112)
music-assistant/music-assistant.github.io: docs/integration/announcements.md

---
title: Announcements
description: Details of the announcements functionality  
---

# Announcements

> Announcements are either Text-to-Speech or audio files that are sent to the MA players via the [`tts.xxxxxx`](https://www.home-assistant.io/integrations/tts/) or [`mass.play_announcement`](../faq/massannounce.md) HA actions. 

Reviewer: ```suggestion
Announcements are either Text-to-Speech or audio files that are sent (from Home Assistant) to the MA players via the [`tts.xxxxxx`](https://www.home-assistant.io/integrations/tts/) or [`mass.play_announcement`](../faq/massannounce.md) HA actions. 
```

![image](../assets/screenshots/announcement-settings.png)

## Native Support

If an announcement is sent to a single player that supports native announcements, the announcement will be handled by the player itself.

Native announcement support usually means that the player can "overlay" the announcement on top of any music that is playing. It will "duck" the volume of the music, play the announcement with a volume boost and then revert the music volume. Music keeps playing and the entire announcement will be handled by the player itself.

Currently the only (known) player type that supports native announcements is Sonos S2 speakers.  Hopefully more player types will follow.

## MA Player Support

If a player does not support native announcements, MA will take care of it by stopping any playing music, adjust the volume (as configured), play the announcement, and then restore the previous state.

!!! note
    The MA announcement feature will ONLY work reliably if the player reports the state (e.g. playing, paused, idle) and the progress report (elapsed time) correctly. If any of those are not updated correctly then announcements will work poorly.

> ESPHome and Snapcast are known to be unreliable with state updates so expect issues with those player types. The same applies to some HA media players.

Reviewer: Can we make this sound less harsh ?
For Snapcast I know @SantiagoSotoC is working on it to imrpove it. Also on the ESPHome side there is a new media player component in the making

Author: ```suggestion
ESPHome and Snapcast can be unreliable with state updates so issues may occur with those player types. The same applies to some HA media players. Future updates of those platforms should improve performance.
```

> If a player does not resume after playing an announcement, check the HTTP profile settings. `Fixed content length` may need to be used. For example, (the current) ESPHome devices have this issue.

Reviewer: ```suggestion
If a player does not resume after playing an announcement, or it takes a very long time, check the HTTP profile settings. `Fixed content length` may need to be used. For example, (the current) ESPHome devices have this issue.
```

## Group Behaviour

> If an announcement is sent to a Native Google Cast Group, Sync Group or a Universal Group, the announcement will be played to ALL child players. The preference to (not) use a pre-announcement sound is defined on the group player. The volume setting of the announcement is defined per player. If one of the child players was already playing, playback will be interrupted and NOT resumed. Bear that in mind with automations and target announcements to the individual players or just restart individual player playback if/when needed. Google Cast groups should play the announcement in sync.

Reviewer: ```suggestion
If an announcement is sent to a native Google Cast Group, Sync Group or a Universal Group, the announcement will always be played to ALL child players. The preference to (not) use a pre-announcement sound is defined on the group player in this case. The volume setting of the announcement is defined per player. If one of the child players was already playing, playback will be interrupted and NOT resumed. Bear that in mind with automations and target announcements to the individual players or just restart individual player playback if/when needed. Google Cast groups should play the announcement in sync.
```

> If an announcement is sent to a group of players that are manually synced, the announcement will be played to all childplayers (in sync)

Reviewer: ```suggestion
If an announcement is sent to a group of players that are manually synced, the announcement will be played to all childplayers (in sync).
```

> If an announcement is sent to a Sync Group (temporary or permananent) of which ALL child players support native announcements, the native functionality will be used to playback the announcement and MA will ensure the announcement is started at the same (millisecond precise) time so all players should play the announcement in sync.

Reviewer: ```suggestion
If an announcement is sent to a Sync Group (temporary or permananent) of which ALL child players support native announcements, the native functionality will be used to playback the announcement and MA will ensure the announcement is started at (more or less) the same time so all players should play the announcement in sync.
```
